spec/observables/from-spec.ts

```diff
@@ -34,6 +34,114 @@ describe('from', () => {
     expect(r).to.throw();
   });
 
+  it('should finalize an AsyncGenerator', (done) => {
+    const results: any[] = [];
+    const sideEffects: any[] = [];
+
+    async function* gen() {
+      try {
+        let i = 0;
+        while (true) {
+          sideEffects.push(i);
+          yield await i++;
+        }
+      } finally {
+        results.push('finalized generator');
+      }
+    }
+
+    const source = from(gen()).pipe(
+      take(3),
+    );
+
+
+    source.subscribe({
+      next: value => results.push(value),
+      complete: () => {
+        results.push('done');
+        setTimeout(() => {
+          try {
+            expect(sideEffects).to.deep.equal([0, 1, 2]);
+            expect(results).to.deep.equal([0, 1, 2, 'done', 'finalized generator']);
+          } finally {
+            done();
+          }
```

Comment:
nitpick: I don't see why a `finally` is needed here. Calling `done` before allowing a failed expectation to bubble out seems unnecessary. AFAICT, with a failed expectation, the test will fail as you'd expect.

Reply:
It was behaving strangely without that try/finally, IIRC. The test would hang when the expectation failed and it wouldn't tell me what failed. I can try removing it and forcing a failure just to confirm.

Reply:
Well, no idea why I added that now. Removed. 🤷‍♂️